spec/inputs/jdbc_spec.rb

```diff
@@ -894,4 +896,82 @@
       plugin.stop
     end
   end
+
+  context "when encoding of some columns need to be changed" do
+
+    let(:settings) {{ "statement" => "SELECT * from test_table" }}
+    let(:events)   { [] }
+    let(:row) do
+      {"column0" => "foo", "column1" => "bar".force_encoding(Encoding::ISO_8859_1), "column2" => 3}
+    end
+
+    before(:each) do
+      allow_any_instance_of(Sequel::JDBC::Derby::Dataset).to receive(:each).and_yield(row)
+      plugin.register
+    end
+
+    after(:each) do
+      plugin.stop
+    end
+
+    it "shuold not convert any column by default" do
+      encoded_row = { "column0" => "foo", "column1" => "bar".force_encoding(Encoding::ISO_8859_1), "column2" => 3 }
+      expect(LogStash::Event).to receive(:new) do |row|
+        row.each do |k, v|
+          next unless v.is_a?(String)
+          expect(row[k].encoding).to eq(encoded_row[k].encoding)
+        end
+      end
+      plugin.run(events)
+    end
+
+    context "when all string columns should be encoded" do
+
+      let(:settings) do
+        { "statement" => "SELECT * from test_table",
+          "charset" => "ISO-8859-1" }
+      end
+
+      let(:row) do
+        {"column0" => "foo".force_encoding(Encoding::ISO_8859_1), "column1" => "bar".force_encoding(Encoding::ISO_8859_1), "column2" => 3}
```

Comment:
cosmetic comment: you format this `let` and the previous `let` in 2 different ways - I would pick one formatting. Personally my preference for multi values hash literals initialization is like this:
```ruby
{
  "column0" => "foo".force_encoding(Encoding::ISO_8859_1),
  "column1" => "bar".force_encoding(Encoding::ISO_8859_1),
  "column2" => 3
}
```